infrastructure/feed/chain/blockheighttracker.go

```diff
@@ -45,27 +48,37 @@ func NewBlockHeightTracker(logger applogger.Logger, client tendermint.Client) *B
 
 func (tracker *BlockHeightTracker) Run() {
 	for {
-		height, err := tracker.client.LatestBlockHeight()
-		if err != nil {
-			tracker.logger.Errorf("error getting chain latest block height: %v", err)
-			<-time.After(1 * time.Second)
-			continue
-		}
+		operation := func() error {
+			height, err := tracker.client.LatestBlockHeight()
+			if err != nil {
+				return fmt.Errorf("error getting chain latest block height: %v", err)
+			}
 
-		for _, subscription := range tracker.subscriptions {
-			select {
-			case subscription <- height:
-			default:
-				tracker.logger.Info("block subscription channel is blocked, maybe busy?")
+			for _, subscription := range tracker.subscriptions {
+				select {
+				case subscription <- height:
+				default:
+					tracker.logger.Info("block subscription channel is blocked, maybe busy?")
+				}
 			}
-		}
 
-		tracker.rwMutex.Lock()
-		tracker.latestBlockHeight = &height
-		tracker.rwMutex.Unlock()
+			tracker.rwMutex.Lock()
+			tracker.latestBlockHeight = &height
+			tracker.rwMutex.Unlock()
 
-		tracker.logger.Infof("updated chain latest block height: %d", height)
-		<-time.After(tracker.pollingInterval)
+			tracker.logger.Infof("updated chain latest block height: %d", height)
+			<-time.After(tracker.pollingInterval)
+			return nil
+		}
+		notifyFn := func(opErr error, backoffDuration time.Duration) {
+			tracker.logger.Errorf("retrying in %s: %v", backoffDuration.String(), opErr)
+		}
+		neverStopExponentialBackoff := backoff.NewExponentialBackOff()
+		neverStopExponentialBackoff.MaxElapsedTime = 0
+		neverStopExponentialBackoff.MaxInterval = 15 * time.Minute
```

Comment:
shall we put this as const on the top for easier configuration?

Reply:
Good point, done